src/ZuluSCSI_disk.cpp

```diff
@@ -500,21 +540,160 @@ static void scsiDiskLoadConfig(int target_idx, const char *section)
     memset(tmp, 0, sizeof(tmp));
     ini_gets(section, "Serial", "", tmp, sizeof(tmp), CONFIGFILE);
     if (tmp[0]) memcpy(img.serial, tmp, sizeof(img.serial));
+
+    if (strlen(section) == 5 && strncmp(section, "SCSI", 4) == 0) // allow within target [SCSIx] blocks only
+    {
+        ini_gets(section, "ImgDir", "", tmp, sizeof(tmp), CONFIGFILE);
+        if (tmp[0])
+        {
+            logmsg("-- SCSI", target_idx, " using image directory \'", tmp, "'");
+            img.image_directory = true;
+        }
+    }
 }
 
-// Check if image file name is overridden in config
-bool scsiDiskGetImageNameFromConfig(image_config_t &img, char *buf, size_t buflen)
+int scsiDiskGetNextImageName(image_config_t &img, char *buf, size_t buflen)
```

Comment:
I think we may want this to be `scsiDiskGetImageNameFromConfig(image_config_t &img, char *buf, size_t buflen, int image_index)` eventually. That would allow e.g. listing all available images, if we add a I2C display or some GUI tool. It does mean a bit more looping and switching to next image is certainly the common case, so I see the point of starting from current file.

I think the code would be easier to read if there was an internal `findNextImageAfter(const char *dirname, const char *filename, char *buf, size_t buflen)`, and then the `scsiDiskGetImageNameFromConfig()` call that.

In any case I'm ok with the code as it is currently written, this can be worked on later.

Reply:
That's a great point regarding future expandability. The only reservation I have with making `image_index` a parameter is that it exposes more of the image switching logic to callers; I had some trouble getting my head wrapped around what pieces of code affect that variable (which I was making more complicated by the parallel `current_image` name tracker).

This approach will need changes if a later PR adds something like the ability to set images via I2C, SCSI, etc. Perhaps the image switching is something that could be moved to a class to protect implementation details?

You're absolutely right that the function had gotten too long and was hard to read: I broke out the actual directory scanning to a new `findNextImageAfter` call. Hopefully that will help a bit.

Reply:
Looks good. I agree that a new class will be good approach in the future.